code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalServiceEventCreate.java

```diff
@@ -106,6 +108,21 @@ void createService(ExternalEvent event, Map<String, Object> serviceData) {
             log.info("Can't process service event. Could not get or create stack. Event: [{}]", event);
             return;
         }
+        
+        List<Service> existingSvcs = objectManager.find(Service.class, 
+            SERVICE.NAME, serviceData.get(FIELD_NAME),
+            SERVICE.REMOVED, null, 
+            SERVICE.STACK_ID, stack.getId(), 
+            SERVICE.ACCOUNT_ID, event.getAccountId(),
+            SERVICE.KIND, ServiceConstants.KIND_KUBERNETES_SERVICE);
+
+        for(Service existingSvc : existingSvcs) {
+            if(existingSvc!=null && !existingSvc.getExternalId().equals(event.getExternalId())) {
+                log.info("Deleting existing kubernetes service {} with externalId {} and accountId {}",
+                            existingSvc.getName(), existingSvc.getExternalId(), existingSvc.getAccountId());
+                objectProcessManager.executeStandardProcess(StandardProcess.REMOVE, existingSvc, null);
```

Comment:
is remove allowed from active? for services. Or should we call deactivate then remove? 
Also it might be a good idea to check for Removing state, and skip those services